engines/myst3/myst3.cpp

```diff
@@ -1505,16 +1507,20 @@ void Myst3Engine::tryAutoSaving() {
 	_lastSaveTime = _system->getMillis();
 
 	// Get a thumbnail of the game screen
-	Common::DisposablePtr<Graphics::Surface, Graphics::SurfaceDeleter> thumbnail(nullptr, DisposeAfterUse::NO);
-	if (_menu->isOpen()) {
-		thumbnail.reset(_menu->borrowSaveThumbnail(), DisposeAfterUse::NO);
-	} else {
-		// Capture thumbnail before saving
-		thumbnail.reset(_menu->captureThumbnail(), DisposeAfterUse::YES);
-	}
+	if (!_menu->isOpen())
+		_menu->generateSaveThumbnail();
 
 	// Open the save file
 	Common::String saveName = "Autosave";
+	saveGame(saveName);
+
+}
+
+void Myst3Engine::saveGame(Common::String saveName) {
+	
+	Common::DisposablePtr<Graphics::Surface, Graphics::SurfaceDeleter> thumbnail(nullptr, DisposeAfterUse::NO);
```

Comment:
There is no need to use `DisposablePtr` here, since ownership remains to the menu (we're just borrowing a read only reference). `const Graphics::Surface *` should be used.